src/trans/gpu/internal/trgtol_mod.F90

```diff
@@ -482,34 +483,44 @@ SUBROUTINE TRGTOL(ALLOCATOR,HTRGTOL,PREEL_REAL,KF_FS,KF_GP,KF_UV_G,KF_SCALARS_G,
         IF(IVSET(JFLD) == ISETV .OR. IVSET(JFLD) == -1 ) THEN
           IFLDS = IFLDS+1
           IF(PRESENT(KPTRGP)) THEN
-            IFLDA(IFLDS)=KPTRGP(JFLD)
+            IFLDAS(IFLDS,INS)=KPTRGP(JFLD)
           ELSE
-            IFLDA(IFLDS)=JFLD
+            IFLDAS(IFLDS,INS)=JFLD
           ENDIF
+        ELSE
+          IFLDAS(JFLD,INS)=IFLDA(JFLD)
```

Comment:
What's the need for this `IF` branch? I don't think `IFLDA` is initialised at this point in the code.

Reply:
I just pushed a commit removing the additional initialization.